Sources/HummingbirdAuth/Authenticator/ClosureAuthenticator.swift

```diff
@@ -15,14 +15,17 @@
 import Hummingbird
 import NIOCore
 
-public struct ClosureAuthenticator<Context: AuthRequestContext, Value: Authenticatable>: AuthenticatorMiddleware {
-    let closure: @Sendable (Request, Context) async throws -> Value?
+public struct ClosureAuthenticator<
+    Identity: Sendable,
+    Context: AuthRequestContext<Identity>
```

Comment:
Is it possible you don't need the Identity generic parameter as that is available via the Context
```swift
ClosureAuthenticator<Context: AuthRequestContext>
```
And then where you need the identity type you use `Context.Identity`